notebooks/Notebook3.md

```diff
@@ -2,33 +2,39 @@
 
 ## Last week's critique
 
-**TODO:** Fill in this part with a summary and reflection on the critique you received for
-last week's work. Answer questions such as:  How, specifically, did the feedback help
-improve the project? Did the feedback point out or offer something you hadn't considered?
-Did it help you make a design decision? Was it helpful in addressing the most pressing
-issues in your project? How will you incorporate the feedback into your work? Will you
-change something about the design, implementation, or evaluation as a result?
+I definitely needed the help this week. Charlie helped on monday and Prof Ben helped on Wednesday. I think we are all dealing with that classic project situation of biting off more then we could chew. Im happy with the progress im making and the changes I can make, but im defintely far from the original goal. 
 
 ## Description
 
-**TODO:** Fill in this part with information about your work this week:
-important design decisions, changes to previous decisions, open questions,
-exciting milestones, preliminary results, etc. Feel free to include images
-(e.g., a sketch of the design or a screenshot of a running program), links to
-code, and any other resources that you think will help clearly convey your
-design process.
+This week I focused on getting the basic version of the parser and functionality for the language done. And by done, I mean 
+the most basic version. So currently a program of the shape: 
+```
+roll name { 
+1d6
+}
+```
+would compile. You could of course change either of the numbers and it would react accordingly. Currently `2dX` evaluates to `1dX + 1dX`. And the name can be any reasonable string. And it works! After some help from prof Ben, I was able to get the system to actually parse this basic code. 
 
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+Now that I have the basic structure complete, I can make progress on implementing *stat* which will be a lot harder, since it needs to start outputing a graph of some capacity. Also checks are something I would like to add but aren't strictly necessary. because right now, equations dont even work, so figuring how to add elements and implement basically math again is key. 
```

Comment:
I'm not totally sure if you still have access to it, but taking a peek at the Expr language we wrote in PLs might be a good idea, as it has a lot of the arithmetic and boolean comparisons that you might want to add for your check function.

Reply:
Have you found any libraries that might help you implement stat yet?